app/k9mail/src/main/java/com/fsck/k9/auth/OAuthConfigurations.kt

```diff
@@ -9,12 +9,31 @@ fun createOAuthConfigurationProvider(): OAuthConfigurationProvider {
         clientId = BuildConfig.OAUTH_GMAIL_CLIENT_ID,
         scopes = listOf("https://mail.google.com/"),
         authorizationEndpoint = "https://accounts.google.com/o/oauth2/v2/auth",
-        tokenEndpoint = "https://oauth2.googleapis.com/token"
+        tokenEndpoint = "https://oauth2.googleapis.com/token",
+        redirectUri = BuildConfig.APPLICATION_ID + ":/oauth2redirect"
+    )
+
+    var outlookConfig = OAuthConfiguration(
+        clientId = BuildConfig.OAUTH_OUTLOOK_CLIENT_ID,
+        scopes = listOf("wl.imap", "wl.emails", "wl.offline_access"),
+        authorizationEndpoint = "https://login.live.com/oauth20_authorize.srf",
+        tokenEndpoint = "https://login.live.com/oauth20_token.srf",
+        redirectUri = BuildConfig.APPLICATION_ID + "://oauth2redirect"
+    )
+
+    var officeConfig = OAuthConfiguration(
+        clientId = BuildConfig.OAUTH_OUTLOOK_CLIENT_ID,
+        scopes = listOf("https://outlook.office365.com/IMAP.AccessAsUser.All", "https://outlook.office365.com/POP.AccessAsUser.All", "https://outlook.office365.com/SMTP.Send", "offline_access"),
+        authorizationEndpoint = "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+        tokenEndpoint = "https://login.microsoftonline.com/common/oauth2/v2.0/token",
+        redirectUri = BuildConfig.APPLICATION_ID + "://oauth2redirect"
     )
 
     return OAuthConfigurationProvider(
         configurations = mapOf(
             listOf("imap.gmail.com", "imap.googlemail.com", "smtp.gmail.com", "smtp.googlemail.com") to googleConfig,
+            listOf("imap-mail.outlook.com", "smtp-mail.outlook.com") to outlookConfig,
```

Comment:
It looks like Thunderbird isn't using `imap-mail.outlook.com` or `smtp-mail.outlook.com` at all. Instead they're only using `outlook.office365.com` and `smtp.office365.com`. Do you know if there are any downsides to that? If not, we should do the same and update the entries in `providers.xml`.

Reply:
`imap-mail.outlook.com` and `smtp-mail.outlook.com` are for Microsoft accounts (outlook.com, live.com, hotmail.com).

I will check again if they work with the `office365.com` servers.
It seems Thunderbird defaults to those.

Reply:
Actually, the outlook.com/live.com/hotmail.com can be used with `office365.com` servers with plain text passwords, but I can not make them work with the OAuth2 endpoints.
Login with an `outlook.com` address on the `login.microsoftonline.com` endpoints with the `outlook.office365.com` scopes fails with this error:
```
You can't sign in here with a personal account. Use your work or school account instead.
```
Using `graph.microsoft.com` in the scopes allows the OAuth2 login to succeed, but the IMAP XOAUTH2 authentication fails later.

Thunderbird does not enable OAuth2 for these email addresses.
See the Mozilla ISP autoconfig database:
- https://github.com/thundernest/autoconfig/blob/master/ispdb/hotmail.com
- https://github.com/thundernest/autoconfig/blob/master/ispdb/outlook.com
- https://github.com/thundernest/autoconfig/blob/master/ispdb/office365.com